proc/common/src/main/java/org/neo4j/gds/StreamProc.java

```diff
@@ -50,6 +50,7 @@ public ComputationResultConsumer<ALGO, ALGO_RESULT, CONFIG, Stream<PROC_RESULT>>
                 NodePropertyValues nodePropertyValues = nodeProperties(computationResult);
                 return LongStream
                     .range(IdMap.START_NODE_ID, graph.nodeCount())
+                    .filter(i -> nodePropertyValues.value(i) != null)
```

Comment:
While this looks very clean, I dont think we want this overhead.

If you look at the implementation of `value()`, you can see that in each invocation we create a new object.
Rather, how about adding a `NodePropertyValues::isValid(long id)` which can be checked instead and avoids allocations?

(also applies to Leiden, Louvain, TriangleCount)